### ISSUE TITLE
Add manuscript to existing code repo

### ISSUE THREAD

**Issue by kescobo** [manubot/rootstock#307]
At the moment, it seems like the setup workflow is only suitable for starting a manuscript in a brand new repository, where the only thing in the repository is the manuscript. It would be nice if the manuscript could live alongside code and other content in a single repository. 

From what I can tell, this may require:

- [ ] decoupling folder structure to avoid clashes (perhaps things that are critical could go in a `manuscript/` or `./manuscript` subfolder)
- [ ] tools for running manuscript CI along with CI for code tests etc

Alternatively, perhaps git submodules could be effective here, and all that would be required is some documentation.

**Comment by agitter** [comment]
@kescobo you're correct that the current setup assumes your repository will only contain the manuscript.  We agree that having the manuscript be more self-contained is desirable.  https://github.com/manubot/template/issues/1 discusses some of that roadmap and how a refactored Manubot could greatly simplify the manuscript repository.

Currently, many projects have a code/analysis repository and a separate manuscript repository.  A recent example:
- [iscb-diversity-manuscript](https://github.com/greenelab/iscb-diversity-manuscript)
- [iscb-diversity code](https://github.com/greenelab/iscb-diversity)

We have also used submodules in the past as you suggested:
- [meta-review manuscript repo](https://github.com/greenelab/meta-review)
- [submodule with deep-review repo for updated statistics](https://github.com/greenelab/meta-review/tree/master/analyses/deep-review-contrib)

**Comment by kescobo** [comment]
Thanks for the links - I'll take a look. The new template does look like a good approach. I'll dig some more and chime in if anything occurs to me 👍

**Comment by dhimmel** [comment]
Yeah. I think the @kescobo touches on an important limitation. Currently, SETUP does not allow adding a manuscript to a pre-existing code repo.

Our plan in the future with `manubot/template` is to reduce Manubot setup to just having the `.github/workflows/manubot.yaml` file in the repo. This would run the GitHub action, which would detect the manuscript location or perhaps read a manubot config file that would tell it where to look and what to do.

This would allow @kescobo to create a manuscript alongside project code. There would be a few restriction like reserving the `gh-pages` branch for Manubot.

For now, I think the best solution would be to create separate repos, since `manubot/template` could still be some ways off.

**Comment by kescobo** [comment]
Thanks @dhimmel. For now, that's what I'm doing, and I'll probably end up going the submodule route. Not done, but https://github.com/Klepac-Ceraj-Lab/resonance1_manuscript

I didn't quite get this set up in time for preprint and submission, but [here's the preprint](https://doi.org/10.1101/2020.02.13.944181)

**Comment by dhimmel** [comment]
> Not done, but https://github.com/Klepac-Ceraj-Lab/resonance1_manuscript

Ah cool! Feel free to add your manuscript to [the catalog source](https://github.com/manubot/catalog/blob/master/catalog.yml) at any point if you'd like it to show up at https://manubot.org/catalog/.

**Comment by agitter** [comment]
I'm cross-referencing https://github.com/manubot/manubot/issues/229#issuecomment-632756812 where there are links to a custom manuscript build process for a repository configured as @kescobo described.  That could be a short term workaround until `manubot/template` is ready.
